x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/ingest/SetSecurityUserProcessor.java

```diff
@@ -81,6 +82,12 @@ public IngestDocument execute(IngestDocument ingestDocument) throws Exception {
                         userObject.put("metadata", user.metadata());
                     }
                     break;
+                case API_KEY_NAME:
+                    Object apiKeyName = authentication.getMetadata().get(ApiKeyService.API_KEY_NAME_KEY);
+                    if (apiKeyName != null) {
+                        userObject.put("api_key_name", apiKeyName);
+                    }
```

Comment:
See my other comment about wanting the id etc as well.
I think this should be a nested object inside the user object.
```
user: {
   username: "...",
   api_key: {
       name: "...", id: "...", "realm": "..."
   }
}   
```